### PR TITLE
Feature/add twin 63

### PR DESCRIPTION
**Pull request by EdwardSafford-NOAA**
This is my first cut at extending the CreatePlot class to allow plotting on a shared x or shared y axis.  It's very incomplete -- I've only tested it with line plots so far, and I haven't completed implementation for the other plot types.  I'd just like some feedback on the basic direction before I proceed.  This is the cleanest way I could see to implement shared axis plotting within the existing class, but I'm entirely open to other ideas.

Unrelated, I'm not clear why the invert_xaxis(), invert_yaxis(), and set_yscale() methods are showing as changes in this PR.  They are in both base:develop and my feature branch from what I can tell.  Any idea what I've messed up there?

### CODE REVIEW

**Comment by CoryMartin-NOAA** [comment]
@EdwardSafford-NOAA did you merge base:develop into your fork:develop? and then merge your fork:develop into this feature branch? This usually happens when there is a mismatch of 'history' between forks/branches.

File: src/emcpy/plots/create_plots.py [hunk 1]
```diff
@@ -194,8 +197,10 @@ def __init__(self, figsize=(8, 6)):
 
         super().__init__(figsize)
         self.ax = self.fig.add_subplot(111)
+        self.shared_ax = None
+        self.shared_ay = None
 
-    def draw_data(self, plot_list):
+    def draw_data(self, plot_list, use_shared_ax=False, use_shared_ay=False):
```
Comment (CoryMartin-NOAA) [inline src/emcpy/plots/create_plots.py]: My general comment is: can we just use `self.shared_ax` and not add these extra function/method arguments? They should be stored in `obj` or `plotobj`, right? Or maybe we need `self.use_shared_ax`, or something like that?

Comment (EdwardSafford-NOAA) [reply src/emcpy/plots/create_plots.py]: Yes, I think that would work.  I'll give it a try.  Thanks.

Comment (EdwardSafford-NOAA) [reply src/emcpy/plots/create_plots.py]: Second cut.  The axes (default ax, optional shared_ax, shared_ay) are properties of the CreatePlot class.  The default, ax, is created when the object is instantiated, the shared axes as needed. 

The plot objects now have a plot_ax variable set to 'ax', 'shared_ax', or 'shared_ay' ('ax' is default).  To do that I added a BasePlot class to emcpy/plots/plots.py which all the other classes now extend.  It's the BasePlot that includes the methods to change the value of plot_ax.  This means the calls to plotting is much more Python-like -- just instantiate all your plot objects, set the desired axes and plot all in one call.

@CoryMartin-NOAA , @kevindougherty-noaa Please take a look when you have a moment.  I still have to work out labels for the second x/y axis, and I have to figure out how to get the legend to work for those figures on shared axes.  I'll hold on that though pending your feedback on this approach.  Thanks.

Comment (CoryMartin-NOAA) [reply src/emcpy/plots/create_plots.py]: From what I saw, this looks great to me, @EdwardSafford-NOAA . The `BasePlot` class is a good addition, and I like the approach of the `_get_axis` call, very neat and clean.

Comment (EdwardSafford-NOAA) [reply src/emcpy/plots/create_plots.py]: Super.  It took me a while to think more OO than procedural; it's been a long time but glad to be back in OO land.  I'll figure out the axis labels for other than the default axes next, and then clean it all up for a proper PR.

**Comment by EdwardSafford-NOAA** [comment]
@CoryMartin-NOAA , @kevindougherty-noaa  I've got the labels on secondary x and y axes working by modifying the CreatePlot.add_x|ylabel() methods.  I have a solution to the legend using the fix described in #63, which places the legend in the figure rather than the ax plot, and doesn't require any code changes.  Adding the legend to the ax plot including all labels on twin() plots is possible but will require more work, and I'd prefer to do that as a separate issue.

I still don't understand what I need to do to correct CreatePlot's invert_xaxis(), invert_yaxis(), and set_yscale() methods from showing as changes.  As you suggest I apparently managed to mess up my history at some point, and that fact the the files in base, fork, and branch are now the same for those lines doesn't seem to matter to git.

**Comment by kevindougherty-noaa** [comment]
@EdwardSafford-NOAA Would you be able to add an example of your new feature in the `tests` directory? That way, Github will automatically make sure that it passes the test that your changes work. Otherwise, I think it looks good.

**Comment by EdwardSafford-NOAA** [comment]
@kevindougherty-noaa Yes, I can take a stab at that.

**Comment by EdwardSafford-NOAA** [comment]
@kevindougherty-noaa I've added a test_plots.py file and eventually got it to pass the coding style check.  Please take a look when you have a moment.  Thanks.

**Review by kevindougherty-noaa — APPROVED** [review]
@EdwardSafford-NOAA Looks great! The examples are good so that people can build off them for their own needs. Maybe in the future I will make a notebook of examples or figure out a way to create a gallery on the doc page!

**Review by CoryMartin-NOAA — APPROVED** [review]
Thanks, @EdwardSafford-NOAA

**Pull request merged**